src/realm/object-store/sync/app_utils.cpp

```diff
@@ -66,9 +56,14 @@ util::Optional<AppError> AppUtils::check_for_errors(const Response& response)
         // ignore parse errors from our attempt to read the error from json
     }
 
+    if (response.client_error_code) {
+        error_msg = (!response.body.empty()) ? response.body : "client error code value considered fatal";
```

Comment:
I know you just moved this around, but i think it'd be clearer as `response.body.empty() ? "client error code.." : response.body`.